### ISSUE TITLE
Unstable fillConvexPoly behavior when all points have the same Y coordinate

### ISSUE THREAD

**Issue by aparpara** [opencv/opencv#9016]
##### System information (version)

- OpenCV => 3.2
- Operating System / Platform => Windows 32 Bit
- Compiler => Visual Studio 2013

##### Detailed description

cv::fillConvexPoly behavior is unstable when the input consists of 3 points having the same Y coordinate. Sometimes it draws a segment as expected, but sometimes it draws a horizontal line throughout the image. In OpenCV 3.1 everything works OK.

##### Steps to reproduce

The program listed below sometimes prints 900 as expected, but sometimes it prints 9100. In this case one can see a horizontal line in the saved image.

```.cpp
#include <opencv2/imgproc.hpp>
#include <opencv2/imgcodecs.hpp>
#include <iostream>

int main()
{
    cv::Mat result(cv::Size(91, 91), CV_8UC1, cv::Scalar(0));
    cv::Point triangle[3] = {
    cv::Point(6, 20), cv::Point(14, 20), cv::Point(13, 20)};
    cv::fillConvexPoly(result, triangle, 3, 100);
    cv::imwrite("result.png", result);
    std::cout << cv::sum(result)[0] << '\n';
  return 0;
}
```

**Comment by alalek** [comment]
Please check the latest code from "master" branch.

**Comment by sturkmen72** [comment]
this should be fixed by  #7812
